### DISCUSSION TITLE
inlet profile with only wall vertex set to U=0

### DISCUSSION THREAD

**Discussion by bigfooted** [su2code/SU2#1613] [General]
Consider a 2D horizontal channel flow where a vertical inlet shares a node with the walls at the top and bottom. The inlet nodes shared by the walls obey the no-slip condition U=0, even when the inlet velocity is given as a constant value over the entire inlet.
Is this (should it be) the same as using an inlet boundary profile where only the wall vertices are set to zero? There seems to be a difference in the radial velocity. See the picture: top: radial velocity when using constant inlet velocity from the config file, bottom: radial velocity when using an inlet profile, where the velocity at the corner nodes is set to 0.
 
![image](https://user-images.githubusercontent.com/7050568/164539371-e26b6ee2-c227-47ba-8ed8-455c2fc6e1aa.png)

**Comment by TobiKattmann** [comment]
Hi Nijso,

The no-slip walls are strongly enforced and the BC_INLET weakly. And the strong boundaries are set after the weak ones.

So I would say the velocity value of the corner node of the inlet should not matter at all, as that gets overruled by the no-slip wall anyway (the respective rows are deleted from the linear system). My expectation would have been to see the very same results here :(  As inlet file and constant inlet from cfg are handled the same way in `BC_INLET`.

I understand that you dont use inlet files for the top picture, but a plain (non-inlet) velocity-inlet. I would be therefore interested in the following results:
- constant inlet-file with the same values as the cfg-only velocity inlet (Set `INC_VELOCITY_INIT` to the values of `MARKER_INLET` for easy creation, then no modifications should be necessary. I would double check the inlet-file though, that the velocity-Magnitude and Direction match with the expected value)
- modified inlet file with the corner nodes set to a 10x bulk-velocity (instead of setting them to zero). Note, that not always the top and bottom rows are also the corner points ;) (but I guess you already considered that in your test as it seems)

In both additional test I would expect the same result as with the inlet file that has zero'ed corners, but lets see :)

Although it should work now, I would work with `DIMENSIONAL` simulations (to be safe).

**Reply by bigfooted** [reply]
Hi Tobi, 
Changing the velocity at the corner nodes has a big impact on the simulation results, see below. inlet velocity=1.0 m/s
Top: corner velocity = 10 m/s
Bottom: corner velocity = 1 m/s (constant inlet velocity, should be same as not using inlet profile)
![screenshot](https://user-images.githubusercontent.com/7050568/164990235-93defa84-293b-44fe-89c0-cc248e71f80c.png)

**Reply by pcarruscag** [reply]
Can you attach some files Nijso?

**Reply by bigfooted** [reply]
I've put this case here: https://github.com/bigfooted/su2cases/tree/master/validation/square

**Reply by pcarruscag** [reply]
After some discussion today, the reason why the profile value matters at the wall is because energy, turbulence, and scalars are not enforced strongly by the wall BC, and thus there is an inlet flux being computed for those quantities using a profile value that does not match the computed wall value.
I still have to find why the momentum is being affected, I expect it has something to do with how pressure and velocity are coupled through the flux schemes.
Nevertheless, we should force consistent values for these intersection nodes.

**Reply by pcarruscag** [reply]
Ok I'm fairly certain this is the cause for the differences in momentum.
The inlet BC is computing a large contribution to the continuity equation (not enforced strongly because we don't know the pressure), this causes a high pressure gradient:
![image](https://user-images.githubusercontent.com/38071223/165778423-4f16a01e-4557-4f83-8618-a5b722ce67a7.png)
Which will affect the momentum in neighboring nodes.

To fix the issue it should be sufficient for BC_Inlet to override the inlet profile values at viscous vertices with values taken directly from the flow solution. This would carry into turbulence and species solvers (which use `GetCharacPrimVar`) and should also fix their problems.

**Reply by TobiKattmann** [reply]
Just a minor clarification in the second to last sentence: viscous vertices = wall vertices ?

**Reply by pcarruscag** [reply]
Viscous-wall vertices, I suppose for a slip wall we want the inlet profile to define the result there.

**Reply by bigfooted** [reply]
ok, just to recap, in bc_inlet we first find if a vertex belongs to a viscous wall. For these vertices we then need to override the vertex value with the flow solution (impose weak solution, obtained from the interior), as @pcarruscag mentions. But isn't that what is already being done for a viscous wall? I

**Reply by pcarruscag** [reply]
It's done for the solution, but not for the "CharacPrimVars" of the marker

**Reply by bigfooted** [reply]
OK, I see, so this is modified through the pointer V_Inlet. 
If on the corner node I set velocity components to zero and the pressure to the value at the normal_neighbor, this problem disappears.     
```
if (jPoint==iPoint){
     for (iDim = 0; iDim < nDim; iDim++)
          V_inlet[iDim+prim_idx.Velocity()] = 0.0;
     V_inlet[prim_idx.Pressure()] = nodes->GetPressure(jPoint);
}
```

**Reply by bigfooted** [reply]
OK, thanks, this issue will be treated in #1631
